### PR TITLE
Added: Remove cache on close

### PR DESCRIPTION
**Pull request by ThaFireDragonOfDeath**
Changes:
Added the feature to clear the cache on exiting Qupzilla.

Sense:
Gives the user more control over disk space and privacy.

Notes:
The following string was added: "Delete cache on close" --> Needs translation.
German: "Cache beim Beenden löschen"

Screenshot:
screenshot-cacheremove

### CODE REVIEW

**Comment by Martii** [comment]
+1 OS X 10.12.3 *(Sierra)*

**Comment by cranes-bill** [comment]
Awesome, thanks!

File: src/lib/app/mainapplication.cpp [hunk 1]
```diff
@@ -745,13 +746,17 @@ void MainApplication::saveSettings()
     if (deleteCookies) {
         m_cookieJar->deleteAllCookies();
     }
+    if (deleteCache) {
+        DataPaths::clearCacheData();
```
Comment (nowrep) [inline src/lib/app/mainapplication.cpp]: This is wrong, correct cache path is `mApp->webProfile()->cachePath()`

File: src/lib/app/mainapplication.cpp [hunk 2]
```diff
@@ -745,13 +746,17 @@ void MainApplication::saveSettings()
...
+    }
 
     m_searchEnginesManager->saveSettings();
     m_plugins->shutdown();
     m_networkManager->shutdown();
 
     DataPaths::clearTempData();
 
+
```
Comment (nowrep) [inline src/lib/app/mainapplication.cpp]: newline

File: src/lib/preferences/preferences.ui [hunk 3]
```diff
@@ -1212,14 +1212,28 @@
...
+                  <widget class="QCheckBox" name="allowCache">
+                   <property name="text">
+                    <string>Allow storing network cache on disk</string>
+                   </property>
+                  </widget>
+                 </item>
+                 <item row="1" column="0" colspan="3">
+                  <widget class="QCheckBox" name="removeCache">
+                   <property name="text">
+                    <string>Delete cache on close</string>
```
Comment (nowrep) [inline src/lib/preferences/preferences.ui]: Indent the QCheckBox with spacer, as with other "Delete on close" options.

**Comment by ThaFireDragonOfDeath** [comment]
Ok, I will correct that...

File: src/lib/app/mainapplication.cpp (hunk 1 above)
Comment (Martii) [inline src/lib/app/mainapplication.cpp]: A change here may affect OS X... did notice it killed the entire `.../qupzilla/cache` folder which I thought was a little odd but it did work. I'll keep my eye out for it.

File: src/lib/app/datapaths.cpp [hunk 4]
```diff
@@ -82,6 +82,12 @@ void DataPaths::clearTempData()
     QzTools::removeDir(path(Temp));
 }
 
+// static
```
Comment (nowrep) [inline src/lib/app/datapaths.cpp]: This can now be removed.

File: src/lib/app/datapaths.cpp [hunk 5]
```diff
@@ -82,6 +82,12 @@ void DataPaths::clearTempData()
...
+void DataPaths::clearCacheData()
+{
+    QzTools::removeDir(path(Cache));
```
Comment (nowrep) [inline src/lib/app/datapaths.cpp]: Same as above.

**Pull request merged**